src/core/manager.c

```diff
@@ -2623,6 +2655,9 @@ static void manager_start_special(Manager *m, const char *name, JobMode mode) {
         const char *s = unit_status_string(job->unit, NULL);
 
         log_info("Activating special unit %s...", s);
+
+        sd_notifyf(false,
+                   "STATUS=Activating special unit %s...", s);
```

Comment:
cast to (void)?

Hmm, thee status string are "sticky", i.e. "systemctl status" will show them forever until another one is posted. Thus, I figure if during normal operation I'd do "systemctl status user@1000.service" it will probably always show me "Activating special unit default.target..." even though it likely long has reached that.

An easy way out might be to reset the string to "Idle" or so one the event queue is empty again.

Reply:
Most `sd_notify()` calls don't have (void)… I think it'd be better to drop them altogether.

> Hmm, thee status string are "sticky", i.e. "systemctl status" will show them forever until another one is posted.

`manager_start_special()` is used for units that generally will end in a demise of the machine: SPECIAL_CTRL_ALT_DEL_TARGET, SPECIAL_EXIT_TARGET, SPECIAL_HALT_TARGET,  SPECIAL_POWEROFF_TARGET, etc, and in those cases the stickiness doesn't matter. It's not used for the default target, afaics.  But you are of course right that if we were left with the status and the machine doesn't go down, that'd be ugly. And at least some of those units do not have to result in the machine going down. I'll add a reset to "Ready". (Not "idle" because we don't want to reset the status to something every time we start some short-lived job, so the "steady state" is not exactly idle…)